### PR TITLE
recipes

### PR DESCRIPTION
**Pull request by doktoric**

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Endpoint.java [hunk 1]
```diff
@@ -11,55 +19,45 @@
...
+@Api(value = "/v4/{workspaceId}/recipes", description = ControllerDescription.RECIPES_V4_DESCRIPTION, protocols = "http,https")
+public interface RecipeV4Endpoint {
 
     @GET
     @Path("")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = RecipeOpDescription.LIST_BY_WORKSPACE, produces = ContentType.JSON, notes = Notes.RECIPE_NOTES,
             nickname = "listRecipesByWorkspace")
-    Set<RecipeViewResponse> listByWorkspace(@PathParam("workspaceId") Long workspaceId);
+    Set<RecipeV4ViewResponse> list(@PathParam("workspaceId") Long workspaceId);
```
Comment (gregito) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Endpoint.java]: so we can use interable object as a return type?
@horadla23 told me that this is not the way what we'd like to follow

Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Endpoint.java]: Use a wrapper class

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/stack/cluster/host/HostGroupToHostGroupResponseConverter.java [hunk 2]
```diff
@@ -59,11 +58,11 @@ public HostGroupResponse convert(HostGroup source) {
         return recipes.stream().map(Recipe::getId).collect(Collectors.toSet());
     }
 
-    private Set<RecipeResponse> getRecipes(Iterable<Recipe> recipes) {
+  /*  private Set<RecipeResponse> getRecipes(Iterable<Recipe> recipes) {
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/stack/cluster/host/HostGroupToHostGroupResponseConverter.java]: remove these lines

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/recipes/RecipeV4RequestToRecipeConverter.java [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
     private MissingResourceNameGenerator missingResourceNameGenerator;
 
     @Override
-    public Recipe convert(RecipeRequest source) {
+    public Recipe convert(RecipeV4Request source) {
         Recipe recipe = new Recipe();
         if (!Strings.isNullOrEmpty(source.getName())) {
             recipe.setName(source.getName());
         } else {
             recipe.setName(missingResourceNameGenerator.generateName(APIResourceType.RECIPE));
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/recipes/RecipeV4RequestToRecipeConverter.java]: Have we any use case when the name is not provided?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/HostGroupDecorator.java [hunk 4]
```diff
@@ -91,13 +90,13 @@ private void prepareRecipesByName(HostGroup subject, Workspace workspace, Collec
...
 
-    private void prepareRecipesByRequests(HostGroup subject, Iterable<RecipeRequest> recipes, Workspace workspace, User user) {
+   /* private void prepareRecipesByRequests(HostGroup subject, Iterable<RecipeRequest> recipes, Workspace workspace, User user) {
         for (RecipeRequest recipe : recipes) {
             Recipe convertedRecipe = conversionService.convert(recipe, Recipe.class);
             convertedRecipe = recipeService.create(convertedRecipe, workspace, user);
             subject.getRecipes().add(convertedRecipe);
         }
-    }
+    }*/
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/HostGroupDecorator.java]: remove the lines

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/CloudbreakTestSuiteInitializer.java [hunk 5]
```diff
@@ -228,12 +228,12 @@ public void cleanUp(@Optional("true") boolean cleanUp) {
...
-                for (Long recipeId : recipeIds) {
-                    cleanUpService.deleteRecipe(cloudbreakClient, recipeId);
-                }
-            }
+//            Set<Long> recipeIds = itContext.getContextParam(CloudbreakITContextConstants.RECIPE_ID, Set.class);
+//            if (recipeIds != null) {
+//                for (Long recipeId : recipeIds) {
+//                    cleanUpService.deleteRecipe(cloudbreakClient, recipeId);
+//                }
+//            }
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/CloudbreakTestSuiteInitializer.java]: remove these lines

File: integration-test/src/main/java/com/sequenceiq/it/util/CleanupService.java [hunk 6]
```diff
@@ -99,9 +99,9 @@ public boolean deleteStack(CloudbreakClient cloudbreakClient, String stackId) {
         return result;
     }
 
-    public void deleteRecipe(CloudbreakClient cloudbreakClient, Long recipeId) {
-        cloudbreakClient.recipeEndpoint().delete(recipeId);
-    }
+   /* public void deleteRecipe(CloudbreakClient cloudbreakClient, Long recipeId) {
+        //cloudbreakClient.recipeV4Endpoint().delete(recipeId);
+    }*/
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/util/CleanupService.java]: remove these lines

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Endpoint.java (hunk 1 above)
Comment (topolyai5) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Endpoint.java]: No, we can't use. @doktoric please change it as well

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Base.java [hunk 7]
```diff
@@ -1,20 +1,23 @@
...
 import com.sequenceiq.cloudbreak.doc.ModelDescriptions.RecipeModelDescription;
 import com.sequenceiq.cloudbreak.validation.ValidBase64;
 
 import io.swagger.annotations.ApiModelProperty;
 
-public abstract class RecipeBase implements JsonEntity {
+public abstract class RecipeV4Base implements JsonEntity {
 
-    @Size(max = 100, min = 5, message = "The length of the recipe's name has to be in range of 5 to 100")
+    @Size(max = 100, min = 5, message = "The length of the recipes's name has to be in range of 5 to 100")
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Base.java]: typo recipes's

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Base.java [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
     @Pattern(regexp = "(^[a-z][-a-z0-9]*[a-z0-9]$)",
-            message = "The recipe's name can only contain lowercase alphanumeric characters and hyphens and has start with an alphanumeric character")
+            message = "The recipes's name can only contain lowercase alphanumeric characters and "
```
Comment (topolyai5) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/recipes/RecipeV4Base.java]: typo recipes's

**Pull request merged**